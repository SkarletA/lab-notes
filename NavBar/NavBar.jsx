/* eslint-disable import/prefer-default-export */
import React, { useState } from 'react';
import { Login } from '../Login/Login';
import style from './NavBar.module.css';
import { Register } from '../Register/Register';

// eslint-disable-next-line react/prop-types
export function NavBar({ onClickButton }) {
  const [openModal, setOpenModal] = useState(false);
  const [openModalRegister, setOpenModalRegister] = useState(false);
  return (
    <nav>
      <ul className={style.navbar}>
        <ul className={style.logoTitle}>
          <img className={style.logo} src="https://svgshare.com/i/g5e.svg" alt="logo" />
          <h1 className={style.titleApp}>BlogSks</h1>
        </ul>
        <li>
          <button
            type="button"
            onClick={() => {
              setOpenModalRegister(true);
              onClickButton();
            }}
            className={style.linkRegister}
          >
            REGISTER
          </button>
        </li>
        <li>
          <button
            type="button"
            onClick={() => {
              setOpenModal(true);
              onClickButton();
            }}
            className={style.linkLogin}
          >
            LOGIN
          </button>
        </li>
      </ul>
      <Login
        openModal={openModal}
        closeModal={() => setOpenModal(false)}
      />
      <Register
        openModalRegister={openModalRegister}
        closeModalRegister={() => setOpenModalRegister(false)}
      />
    </nav>
  );
}
