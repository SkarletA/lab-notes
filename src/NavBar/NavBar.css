ul {
  margin-block-start: 0;
  margin-block-end: 0;
  padding-inline-start: 0;
}
.navbar {
  align-items: center;
  background-color: #7D1935;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 0;
}
.navbar a:hover {
  background-color: black;
  color: bisque;
}
.logo {
  height: 3em;
  margin-right: 0.5em;
  margin-top: 0.5em;
  width: 3em;
}
.logo-title {
  display: flex;
  flex-direction: row;
  
}
.title-app {
  color: white;
  font-size: 1.5em;
  font-weight: bold;
}
li {
  list-style:none;
  font-weight: bold;
}
.link-register, .link-login {
  text-decoration: none;
  color: white;
  cursor: pointer;
  background-color: #7D1935;
  border: none;
}

