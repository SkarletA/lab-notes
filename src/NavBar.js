import React from "react";
import './NavBar.css';

// eslint-disable-next-line jsx-a11y/anchor-is-valid
export function NavBar() {
  // const [login, setlogin] = useState(false);
  // const toggleLogin = () => {
  //   setlogin();
  // }
  return (
    <ul className="navbar">
      <ul className="logo-title">
        <img className="logo" src="https://svgshare.com/i/g5e.svg" alt="logo"/>
        <h1 className="title-app">BlogSks</h1>
      </ul>
      <li><button className="btn-register">REGISTER</button></li>
      <li><button className="btn-login">LOGIN</button></li>
    </ul>

  );


}