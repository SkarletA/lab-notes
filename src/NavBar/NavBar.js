import React, {useState} from "react";
import { Login } from "../Login/Login";
import './NavBar.css';
import { Register } from '../Register/Register';

// eslint-disable-next-line jsx-a11y/anchor-is-valid
export function NavBar() {
  const [openModal, setOpenModal] = useState(false);
  const [openModalRegister, setOpenModalRegister] = useState(false);
  
  
  return (
    <>
    <ul className="navbar">
      <ul className="logo-title">
        <img className="logo" src="https://svgshare.com/i/g5e.svg" alt="logo"/>
        <h1 className="title-app">BlogSks</h1>
      </ul>
      <li><button onClick={()=> {setOpenModalRegister(true)}} className="link-register">REGISTER</button></li>
      <li><button
      onClick={()=> {setOpenModal(true)}} className="link-login">LOGIN</button></li>
    </ul>
    <Login openModal={openModal} closeModal={()=> setOpenModal(false)}/>
    <Register openModalRegister={openModalRegister} closeModalRegister={()=> setOpenModalRegister(false)}/>
    </>
    

  );


}