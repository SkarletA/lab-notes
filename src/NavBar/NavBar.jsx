/* eslint-disable import/prefer-default-export */
import React, { useState } from 'react';
import { Login } from '../Login/Login';
import './NavBar.css';
import { Register } from '../Register/Register';

// eslint-disable-next-line jsx-a11y/anchor-is-valid
export function NavBar() {
  const [openModal, setOpenModal] = useState(false);
  const [openModalRegister, setOpenModalRegister] = useState(false);
  return (
    <nav>
      <ul className="navbar">
        <ul className="logo-title">
          <img className="logo" src="https://svgshare.com/i/g5e.svg" alt="logo" />
          <h1 className="title-app">BlogSks</h1>
        </ul>
        <li>
          <button
            type="button"
            onClick={() => { setOpenModalRegister(true); }}
            className="link-register"
          >
            REGISTER
          </button>
        </li>
        <li>
          <button
            type="button"
            onClick={() => { setOpenModal(true); }}
            className="link-login"
          >
            LOGIN
          </button>
        </li>
      </ul>
      <Login
        openModal={openModal}
        closeModal={() => setOpenModal(false)}
      />
      <Register
        openModalRegister={openModalRegister}
        closeModalRegister={() => setOpenModalRegister(false)}
      />
    </nav>
  );
}
