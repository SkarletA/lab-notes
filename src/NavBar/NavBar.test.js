/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { NavBar } from './NavBar';

describe(('Test component NavBar'), () => {
  test('render component NavBar', () => {
    let navbar;
    act(() => {
      navbar = create(
        <Router>
          <NavBar />
        </Router>,
      );
      expect(navbar.toJSON()).toMatchSnapshot();
    });
  });

  test('create a element dom', () => {
    const testRender = create(
      <Router>
        <NavBar />
      </Router>,
    );
    const testInstance = testRender.root;
    expect(testInstance.findByProps({ className: 'titleApp' }).children).toEqual(['BlogSks']);
  });

  const mockOnClick = jest.fn();

  test('value when clicked button Login', () => {
    const componentNav = render(
      <Router>
        <NavBar onClickButton={mockOnClick} />
      </Router>,
    );

    const button = componentNav.getByText('LOGIN');
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
  test('value when clicked button Register', () => {
    const componentNav = render(
      <Router>
        <NavBar onClickButton={mockOnClick} />
      </Router>,
    );

    const button = componentNav.getByText('REGISTER');
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
