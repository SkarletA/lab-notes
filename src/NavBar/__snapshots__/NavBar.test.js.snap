// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component NavBar render component NavBar 1`] = `null`;
